core/prepare.go

```diff
@@ -64,12 +64,16 @@ func makePrepareValidator(n uint32, verifyUI uiVerifier, validateRequest request
 
 // makePrepareApplier constructs an instance of prepareApplier using
 // id as the current replica ID, and the supplied abstract interfaces.
-func makePrepareApplier(id uint32, prepareSeq requestSeqPreparer, collectCommitment commitmentCollector, handleGeneratedUIMessage generatedUIMessageHandler) prepareApplier {
+func makePrepareApplier(id uint32, prepareSeq requestSeqPreparer, collectCommitment commitmentCollector, handleGeneratedUIMessage generatedUIMessageHandler, startReqTimer requestTimerStarter, stopPrepTimer prepareTimerStopper) prepareApplier {
 	return func(prepare *messages.Prepare) error {
-		if new := prepareSeq(prepare.Msg.Request); !new {
+		request := prepare.Msg.Request
+		if new := prepareSeq(request); !new {
 			return fmt.Errorf("Request already prepared")
 		}
 
+		stopPrepTimer(prepare.Msg.ReplicaId)
+		startReqTimer(request.Msg.ClientId, prepare.Msg.View)
```

Comment:
The request timer is started by `requestApplier`, I think there is no need to restart the request timer here.

Reply:
Applied, thank you.